daemon/api_systems.go

```diff
@@ -156,29 +157,42 @@ func getSystemDetails(c *Command, r *http.Request, user *auth.UserState) Respons
 	return SyncResponse(rsp)
 }
 
-type systemActionRequest struct {
-	Action string `json:"action"`
-	client.SystemAction
-}
-
 func postSystemsAction(c *Command, r *http.Request, user *auth.UserState) Response {
-	var req systemActionRequest
 	systemLabel := muxVars(r)["label"]
 
-	decoder := json.NewDecoder(r.Body)
-	if err := decoder.Decode(&req); err != nil {
-		return BadRequest("cannot decode request body into system action: %v", err)
+	// XXX: is there a better way to partially decode?
```

Comment:
in practice you could put all the fields in one struct, they seem unlikely to clash at least medium term

Reply:
I reworked this a bit now to have it all in a single struct.